code/datums/components/slippery.dm

```diff
@@ -0,0 +1,57 @@
+/**
+  * # Slip Component
+  *
+  * This is a component that can be applied to any movable atom (mob or obj).
+  *
+  * While the atom has this component, any human mob that walks over it will have a chance to slip.
+  * Duration, tiles moved, and so on, depend on what variables are passed in when the component is added.
+  *
+  */
+/datum/component/slippery
+	/// Text that gets displayed in the slip proc, i.e. "user slips on [description]"
+	var/description
+	/// The amount of stun to apply after slip.
+	var/stun
+	/// The amount of weaken to apply after slip.
+	var/weaken
+	/// The chance that walking over the parent will slip you.
+	var/slip_chance
+	/// The amount of tiles someone will be moved after slip.
+	var/slip_tiles
+	/// TRUE If this slip can be avoided by walking.
+	var/walking_is_safe
+	/// TRUE if having no slip shoes makes you immune to this slip.
+	var/noslip_is_immune
+	/// The verb that players will see when someone slips on the parent. In the form of "You [slip_verb]ped on".
+	var/slip_verb
+
+/datum/component/slippery/Initialize(_description, _stun = 0, _weaken = 0, _slip_chance = 100, _slip_tiles = 0, \
+									_walking_is_safe = TRUE, _noslip_is_immune = TRUE, _slip_verb = "slip")
```

Comment:
Just put this all on teh same line, please.

Reply:
Sure